Port few shot learning with reptile

anas-rz opened the pull request:
Update imports

fchollet reviewed (approved):
LGTM -- do you have a link to the notebook and the git diff?

anas-rz commented:
[COLAB NOTEBOOK](https://colab.research.google.com/drive/1zKr98zIqk2VMfsu0W5gYMfPLkUWoGpay?usp=sharing)
Here it is
```
diff --git a/reptile.py b/reptile_keras_core.py
index c711378..ddf5ef9 100644
--- a/reptile.py
+++ b/reptile_keras_core.py
@@ -1,8 +1,9 @@
 """
 Title: Few-Shot learning with Reptile
 Author: [ADMoreau](https://github.com/ADMoreau)
+Converted to Keras Core By: [Muhammad Anas Raza](https://anasrz.com)
 Date created: 2020/05/21
-Last modified: 2020/05/30
+Last modified: 2023/07/20
 Description: Few-shot classification on the Omniglot dataset using Reptile.
 Accelerator: GPU
 """
@@ -17,13 +18,16 @@ The algorithm works by performing Stochastic Gradient Descent using the
 difference between weights trained on a mini-batch of never-seen-before data and the
 model weights prior to training over a fixed number of meta-iterations.
 """
+import os
+os.environ["KERAS_BACKEND"] = "tensorflow"
+
+import keras_core as keras
+from keras_core import layers
 
 import matplotlib.pyplot as plt
 import numpy as np
 import random
 import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
 import tensorflow_datasets as tfds
 
 """
@@ -41,7 +45,7 @@ eval_iters = 5
 inner_iters = 4
 
 eval_interval = 1
-train_shots = 20
+train_shots = 1
 shots = 5
 classes = 5
 
@@ -68,7 +72,9 @@ class Dataset:
         # Download the tfrecord files containing the omniglot data and convert to a
         # dataset.
         split = "train" if training else "test"
-        ds = tfds.load("omniglot", split=split, as_supervised=True, shuffle_files=False)
+        ds = tfds.load(
+            "omniglot", split=split, as_supervised=True, shuffle_files=False
+        )
         # Iterate over the dataset to get each individual image and its class,
         # and put that data into a dictionary.
         self.data = {}
@@ -222,7 +228,9 @@ for meta_iter in range(meta_iters):
             for images, labels in train_set:
                 with tf.GradientTape() as tape:
                     preds = model(images)
-                    loss = keras.losses.sparse_categorical_crossentropy(labels, preds)
+                    loss = keras.losses.sparse_categorical_crossentropy(
+                        labels, preds
+                    )
                 grads = tape.gradient(loss, model.trainable_weights)
                 optimizer.apply_gradients(zip(grads, model.trainable_weights))
             test_preds = model.predict(test_images)
@@ -235,7 +243,8 @@ for meta_iter in range(meta_iters):
         testing.append(accuracies[1])
         if meta_iter % 100 == 0:
             print(
-                "batch %d: train=%f test=%f" % (meta_iter, accuracies[0], accuracies[1])
+                "batch %d: train=%f test=%f"
+                % (meta_iter, accuracies[0], accuracies[1])
             )
 
 """
@@ -245,7 +254,9 @@ for meta_iter in range(meta_iters):
 # First, some preprocessing to smooth the training and testing arrays for display.
 window_length = 100
 train_s = np.r_[
-    training[window_length - 1 : 0 : -1], training, training[-1:-window_length:-1]
+    training[window_length - 1 : 0 : -1],
+    training,
+    training[-1:-window_length:-1],
 ]
 test_s = np.r_[
     testing[window_length - 1 : 0 : -1], testing, testing[-1:-window_length:-1]

```

fchollet commented:
Thanks -- looking good!
